diamond-docs/build/build

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+set -ex
+
+# Create the virtualenv
+virtualenv venv
+. venv/bin/activate
+
+# Define and ensure the PIP cache
+PIP_SDIST_INDEX="$HOME/.cache/pip"
+mkdir -p $PIP_SDIST_INDEX
+
+# Install the package by trying with the cache first, otherwise doing a download only, and then
+# trying to install from the cache again.
+if ! venv/bin/pip install --find-links="file://$PIP_SDIST_INDEX" --no-index tox; then
+    venv/bin/pip install --download-directory="$PIP_SDIST_INDEX" tox
+    venv/bin/pip install --find-links="file://$PIP_SDIST_INDEX" --no-index tox
+fi
```

Comment:
most of the contents on this script are already available for you if you are pulling it in the shell builder with:

    - ../../../scripts/build_utils.sh

If you do that, then this script can be like the ceph-installer: https://github.com/ceph/ceph-build/blob/master/ceph-installer-docs/build/build